### PR TITLE
Fix sc statuses not loaded on time when under heavy load

### PR DESCRIPTION
**Pull request by csnv**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Added a new flag to indicate sc data has been loaded and postpone clif->parse_LoadEndAck until all requirements are met.

When under heavy load, pc->reg_received runs before pc->scdata_received, thus invoking character calculation without taking any sc statuses like SC_PUSH_CART into consideration.

Along with incorrect character stats or substats, this issue makes the player not able to see visual effects or elements related to statuses, as well as the cart, until the player changes maps or uses a fly wing.

**Issues addressed:** 

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

File: src/map/pc.c [hunk 1]
```diff
@@ -12260,14 +12260,19 @@ static void pc_scdata_received(struct map_session_data *sd)
 		pc->expire_check(sd);
 	}
 
-	if( sd->state.standalone ) {
+
+	if (sd->sc.data[SC_SOULENERGY] != NULL)
```
Comment (skyleo) [inline src/map/pc.c:12264]: nit: This seems unrelated to the commit it's in. Split into individual commit.

Comment (csnv) [reply src/map/pc.c:12264]: There's no additional logic here, I reordered it so clif->pLoadEndAck is called last.

Comment (skyleo) [reply src/map/pc.c:12264]: Hmmm. okay. I was looking at it more in the terms of you moved SC_SOULENERGY check.

**Review by EPuncker — APPROVED** [review]

File: src/map/pc.c [hunk 2]
```diff
@@ -12260,14 +12260,19 @@ static void pc_scdata_received(struct map_session_data *sd)
...
+		sd->soulball = sd->sc.data[SC_SOULENERGY]->val1;
+
+	sd->state.scloaded = 1;
+	if (sd->state.standalone) {
 		clif->pLoadEndAck(0,sd);
 		pc->autotrade_populate(sd);
 		pc->autotrade_start(sd);
+	} else {
+		clif->pLoadEndAck(sd->fd, sd);
```
Comment (MishimaHaruna) [inline src/map/pc.c:12273]: I'm not sure if I'm overlooking it, but I don't think I can see any mechanism that prevent loadendack execution if the client isn't ready yet (i.e. this should probably be called from pc_scdata_received if the client aleady triggered LoadEndAck and its execution was deferred bythe `sd->state.scloaded == 0` condition?)

**Review by MishimaHaruna — CHANGES_REQUESTED** [review]

File: src/map/pc.c (hunk 2 above)
Comment (csnv) [reply src/map/pc.c:12273]: That's exactly the case in this change. pLoadEndAck will be called but under normal circumstances not run entirely until the client has loaded, aka `sd->state.loaded == 1`

The aim of this change is only for those rare cases where the server is under heavy stress and the client loads faster than the server can receive SC data.

Comment (MishimaHaruna) [reply src/map/pc.c:12273]: Ok, it looks fine. I was worried about the execution of `connect_new = 1` before the `pc_regs_received` call, but in the code paths I examined it doesn't seem to cause any side effects. And the same mechanisms that prevent double execution through `pc_regs_received` seem to be adequate to prevent double execution through `pc_scdata_received`

**Review by MishimaHaruna — APPROVED** [review]
